src/napari_threedee/data_models/surfaces.py

```diff
@@ -0,0 +1,176 @@
+import os
+from typing import List
+
+import napari
+import numpy as np
+import pandas as pd
+import zarr
+from pydantic import BaseModel, validator
+
+from napari_threedee.annotators.base import N3dDataModel
+from napari_threedee.annotators.constants import N3D_METADATA_KEY, ANNOTATION_TYPE_KEY
+from napari_threedee.annotators.surfaces.constants import LEVEL_ID_FEATURES_KEY, \
+    SURFACE_ID_FEATURES_KEY, SURFACE_ANNOTATION_TYPE_KEY, COLOR_CYCLE
+from napari_threedee.annotators.surfaces.validation import validate_layer, \
+    validate_n3d_zarr
+
+
+class N3dSurface(BaseModel):
+    data: List[np.ndarray]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @property
+    def ndim(self) -> int:
+        return self.data[0].shape[-1]
+
+    @validator('data', pre=True)
+    def ensure_float32_ndarray_list(cls, value):
+        data = [
+            np.asarray(arr, dtype=np.float32)
+            for arr
+            in value
+        ]
+        return data
+
+    @property
+    def n_points(self) -> int:
+        return sum(len(arr) for arr in self.data)
+
+    @property
+    def n_levels(self) -> int:
+        return len(self.data)
+
+    @property
+    def points(self) -> np.ndarray:
+        return np.concatenate([level for level in self.data])
+
+    @property
+    def level_ids(self) -> np.ndarray:
+        level_ids = [
+            [idx] * len(level)
+            for idx, level
+            in enumerate(self.data)
+        ]
+        return np.concatenate(level_ids)
+
+    def __iter__(self) -> np.ndarray:
+        yield from self.data
+
+
+class N3dSurfaces(N3dDataModel):
```

Comment:
What do you think about `N3dSurfaceCollection`? I think `N3dSurfaces` is too similar to `N3dSurface`. I'm open to other names, but I do think we should choose something a bit more different,.